esp/services/ws_store/espstorelib/daliKVStore.cpp

```diff
@@ -57,6 +59,11 @@ bool CDALIKVStore::createStore(const char * apptype, const char * storename, con
     if (maxvalsize != 0)
         apptree->setPropInt(DALI_KVSTORE_MAXVALSIZE_ATT, maxvalsize);
 
+    Owned<IRemoteConnection> conn = querySDS().connect(DALI_KVSTORE_PATH, myProcessSession(), RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT_KVSTORE);
```

Comment:
I think this introduces a potential problem (via a race condition).
Multiple clients could reach this point, having all performed the read lock at line 28, and found the 'Store' not to exist. Then all perform this write lock (in sequence), and all perform the addPropTree, resulting in multiple ambiguous copies of the same named store.

The code will need to recheck after the write (exclusive) lock is established I think.
Also, it would be better to keep the existing connection (not release it), and use changeMode to switch from RTM_LOCK_READ to RTM_LOCK_WRITE, e.g.:
 initially establish the read lock, check for the store's existence, then do a:
```
conn->changeMode(RTM_LOCK_WRITE);
```